Issue: continue-on-error within step is not respected

<!--
- Make sure you are able to reproduce it on the [latest version](https://github.com/nektos/act/releases)
- Search the existing issues.
- Refer to [README](https://github.com/nektos/act/blob/master/README.md).
-->

## Act version

<!-- Paste output of `act --version` -->
act version 0.2.20
```none

```

## Expected behaviour

<!-- Describe how whole process should go and finish -->
When a job has a step containing `continue-on-error: true` and that step fails, the job should continue as specified by https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error

## Actual behaviour

<!-- Describe what happened -->
When a job has a step containing `continue-on-error: true` and that step fails, the job stops at that step

## Workflow and/or repository

<details>
  <summary>workflow</summary>

```none
# This workflow will run tests on PRs

name: PyTest On PR

on:
  pull_request:
    branches:
      - master

jobs:
  test:
    name: Install dependencies and test
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Set up Python
        uses: actions/setup-python@v2.2.1
        with:
          python-version: "3.6"
      - name: Install dependencies
        run: make install
      - name: Run tests
        run: make test TEST_FLAGS="-n 5"
      - name: Run MyPy
        run: make mypy ONLY_MODIFIED=0
        continue-on-error: true
      - name: Timing test
        timeout-minutes: 2
        run: python -m scripts.statcast_timing
```

https://github.com/TheCleric/pybaseball

</details>

## Steps to reproduce

<!--
Make sure to include command you used to run `act`
e.g.:
1. Clone example repo (https://github.com/cplee/github-actions-demo)
2. Enter cloned repo directory
3. Run `act -s SUPER_SECRET=im-a-value`
-->

1. Clone https://github.com/TheCleric/pybaseball
2. Enter cloned repo directory
3. Checkout the `feature/remove_altair` branch (`git checkout origin/feature/remove_altair`)
3. Run `act pull_request -P ubuntu-latest=nektos/act-environments-ubuntu:18.04` 

## `act` output

<!-- Paste output from your terminal, use `-v` or `--verbose` for richer output -->

<details>
  <summary>Log</summary>

```none
[PyTest On PR/Install dependencies and test] 🚀  Start image=nektos/act-environments-ubuntu:18.04
[PyTest On PR/Install dependencies and test]   🐳  docker run image=nektos/act-environments-ubuntu:18.04 entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[PyTest On PR/Install dependencies and test]   🐳  docker cp src=/Users/adam.weeden/Dev/pybaseball/. dst=/github/workspace
[PyTest On PR/Install dependencies and test] ⭐  Run actions/checkout@v2
[PyTest On PR/Install dependencies and test]   ✅  Success - actions/checkout@v2
[PyTest On PR/Install dependencies and test] ⭐  Run Set up Python
[PyTest On PR/Install dependencies and test]   ☁  git clone 'https://github.com/actions/setup-python' # ref=v2.2.1
[PyTest On PR/Install dependencies and test]   🐳  docker cp src=/Users/adam.weeden/.cache/act/actions-setup-python@v2.2.1 dst=/actions/
[PyTest On PR/Install dependencies and test]   💬  ::debug::Semantic version spec of 3.6 is 3.6
[PyTest On PR/Install dependencies and test]   💬  ::debug::isExplicit: 
[PyTest On PR/Install dependencies and test]   💬  ::debug::explicit? false
[PyTest On PR/Install dependencies and test]   💬  ::debug::isExplicit: 2.7.17
[PyTest On PR/Install dependencies and test]   💬  ::debug::explicit? true
[PyTest On PR/Install dependencies and test]   💬  ::debug::isExplicit: 3.5.9
[PyTest On PR/Install dependencies and test]   💬  ::debug::explicit? true
[PyTest On PR/Install dependencies and test]   💬  ::debug::isExplicit: 3.6.10
[PyTest On PR/Install dependencies and test]   💬  ::debug::explicit? true
[PyTest On PR/Install dependencies and test]   💬  ::debug::isExplicit: 3.7.6
[PyTest On PR/Install dependencies and test]   💬  ::debug::explicit? true
[PyTest On PR/Install dependencies and test]   💬  ::debug::isExplicit: 3.8.1
[PyTest On PR/Install dependencies and test]   💬  ::debug::explicit? true
[PyTest On PR/Install dependencies and test]   💬  ::debug::evaluating 5 versions
[PyTest On PR/Install dependencies and test]   💬  ::debug::matched: 3.6.10
[PyTest On PR/Install dependencies and test]   💬  ::debug::checking cache: /opt/hostedtoolcache/Python/3.6.10/x64
[PyTest On PR/Install dependencies and test]   💬  ::debug::Found tool in cache Python 3.6.10 x64
[PyTest On PR/Install dependencies and test]   ⚙  ::add-path:: /opt/hostedtoolcache/Python/3.6.10/x64
[PyTest On PR/Install dependencies and test]   ⚙  ::add-path:: /opt/hostedtoolcache/Python/3.6.10/x64/bin
[PyTest On PR/Install dependencies and test]   ⚙  ::set-output:: python-version=3.6.10
| Successfully setup CPython (3.6.10)
[PyTest On PR/Install dependencies and test]   ❓  ##[add-matcher]/actions/actions-setup-python@v2.2.1/.github/python.json
[PyTest On PR/Install dependencies and test]   ✅  Success - Set up Python
[PyTest On PR/Install dependencies and test] ⭐  Run Install dependencies
| pip install --upgrade pip; pip install -e .[test]
| Collecting pip
|   Downloading pip-21.0.1-py3-none-any.whl (1.5 MB)
     |################################| 1.5 MB 1.1 MB/s 
| Installing collected packages: pip
|   Attempting uninstall: pip
|     Found existing installation: pip 20.0.2
|     Uninstalling pip-20.0.2:
|       Successfully uninstalled pip-20.0.2
| Successfully installed pip-21.0.1
| Obtaining file:///github/workspace
|   Installing build dependencies ... done
|   Getting requirements to build wheel ... done
|     Preparing wheel metadata ... done
| Collecting pandas>=1.0.3
|   Downloading pandas-1.1.5-cp36-cp36m-manylinux1_x86_64.whl (9.5 MB)
     |################################| 9.5 MB 5.4 MB/s 
| Collecting scipy>=1.4.0
|   Downloading scipy-1.5.4-cp36-cp36m-manylinux1_x86_64.whl (25.9 MB)
     |################################| 25.9 MB 17.8 MB/s 
| Collecting pygithub>=1.51
|   Downloading PyGithub-1.54.1-py3-none-any.whl (289 kB)
     |################################| 289 kB 19.5 MB/s 
| Collecting fuzzywuzzy[speedup]>=0.15.0
|   Downloading fuzzywuzzy-0.18.0-py2.py3-none-any.whl (18 kB)
| Collecting numpy>=1.13.0
|   Downloading numpy-1.19.5-cp36-cp36m-manylinux2010_x86_64.whl (14.8 MB)
     |################################| 14.8 MB 18.1 MB/s 
| Collecting tqdm>=4.50.0
|   Downloading tqdm-4.56.2-py2.py3-none-any.whl (72 kB)
     |################################| 72 kB 2.3 MB/s 
| Collecting matplotlib>=2.0.0
|   Downloading matplotlib-3.3.4-cp36-cp36m-manylinux1_x86_64.whl (11.5 MB)
     |################################| 11.5 MB 15.9 MB/s 
| Collecting lxml>=4.2.1
|   Downloading lxml-4.6.2-cp36-cp36m-manylinux1_x86_64.whl (5.5 MB)
     |################################| 5.5 MB 22.7 MB/s 
| Collecting requests>=2.18.1
|   Downloading requests-2.25.1-py2.py3-none-any.whl (61 kB)
     |################################| 61 kB 12.3 MB/s 
| Collecting beautifulsoup4>=4.4.0
|   Downloading beautifulsoup4-4.9.3-py3-none-any.whl (115 kB)
     |################################| 115 kB 13.9 MB/s 
| Collecting pyarrow>=1.0.1
|   Downloading pyarrow-3.0.0-cp36-cp36m-manylinux2014_x86_64.whl (20.7 MB)
     |################################| 20.7 MB 13.4 MB/s 
| Collecting mypy>=0.782
|   Downloading mypy-0.800-cp36-cp36m-manylinux2010_x86_64.whl (21.6 MB)
     |################################| 21.6 MB 26.9 MB/s 
| Collecting pytest>=6.0.2
|   Downloading pytest-6.2.2-py3-none-any.whl (280 kB)
     |################################| 280 kB 15.1 MB/s 
| Collecting pytest-cov>=2.10.1
|   Downloading pytest_cov-2.11.1-py2.py3-none-any.whl (20 kB)
| Collecting pytest-xdist>=2.1.0
|   Downloading pytest_xdist-2.2.1-py3-none-any.whl (37 kB)
| Collecting soupsieve>1.2
|   Downloading soupsieve-2.2-py3-none-any.whl (33 kB)
| Collecting python-levenshtein>=0.12
|   Downloading python-Levenshtein-0.12.2.tar.gz (50 kB)
     |################################| 50 kB 9.6 MB/s 
| Collecting pillow>=6.2.0
|   Downloading Pillow-8.1.0-cp36-cp36m-manylinux1_x86_64.whl (2.2 MB)
     |################################| 2.2 MB 859 kB/s 
| Collecting python-dateutil>=2.1
|   Downloading python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
     |################################| 227 kB 10.7 MB/s 
| Collecting pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3
|   Downloading pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
     |################################| 67 kB 3.0 MB/s 
| Collecting cycler>=0.10
|   Downloading cycler-0.10.0-py2.py3-none-any.whl (6.5 kB)
| Collecting kiwisolver>=1.0.1
|   Downloading kiwisolver-1.3.1-cp36-cp36m-manylinux1_x86_64.whl (1.1 MB)
     |################################| 1.1 MB 17.8 MB/s 
| Collecting six
|   Downloading six-1.15.0-py2.py3-none-any.whl (10 kB)
| Collecting typed-ast<1.5.0,>=1.4.0
|   Downloading typed_ast-1.4.2-cp36-cp36m-manylinux1_x86_64.whl (743 kB)
     |################################| 743 kB 13.7 MB/s 
| Collecting typing-extensions>=3.7.4
|   Downloading typing_extensions-3.7.4.3-py3-none-any.whl (22 kB)
| Collecting mypy-extensions<0.5.0,>=0.4.3
|   Downloading mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
| Collecting pytz>=2017.2
|   Downloading pytz-2021.1-py2.py3-none-any.whl (510 kB)
     |################################| 510 kB 16.0 MB/s 
| Collecting deprecated
|   Downloading Deprecated-1.2.11-py2.py3-none-any.whl (9.1 kB)
| Collecting pyjwt<2.0
|   Downloading PyJWT-1.7.1-py2.py3-none-any.whl (18 kB)
| Collecting py>=1.8.2
|   Downloading py-1.10.0-py2.py3-none-any.whl (97 kB)
     |################################| 97 kB 10.1 MB/s 
| Collecting packaging
|   Downloading packaging-20.9-py2.py3-none-any.whl (40 kB)
     |################################| 40 kB 8.5 MB/s 
| Collecting toml
|   Downloading toml-0.10.2-py2.py3-none-any.whl (16 kB)
| Collecting attrs>=19.2.0
|   Downloading attrs-20.3.0-py2.py3-none-any.whl (49 kB)
     |################################| 49 kB 10.2 MB/s 
| Collecting pluggy<1.0.0a1,>=0.12
|   Downloading pluggy-0.13.1-py2.py3-none-any.whl (18 kB)
| Collecting iniconfig
|   Downloading iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
| Collecting importlib-metadata>=0.12
|   Downloading importlib_metadata-3.4.0-py3-none-any.whl (10 kB)
| Collecting zipp>=0.5
|   Downloading zipp-3.4.0-py3-none-any.whl (5.2 kB)
| Collecting coverage>=5.2.1
|   Downloading coverage-5.4-cp36-cp36m-manylinux2010_x86_64.whl (240 kB)
     |################################| 240 kB 20.4 MB/s 
| Collecting execnet>=1.1
|   Downloading execnet-1.8.0-py2.py3-none-any.whl (39 kB)
| Collecting pytest-forked
|   Downloading pytest_forked-1.3.0-py2.py3-none-any.whl (4.7 kB)
| Collecting apipkg>=1.4
|   Downloading apipkg-1.5-py2.py3-none-any.whl (4.9 kB)
| Requirement already satisfied: setuptools in /opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages (from python-levenshtein>=0.12->fuzzywuzzy[speedup]>=0.15.0->pybaseball==2.1.1) (40.6.2)
| Collecting certifi>=2017.4.17
|   Downloading certifi-2020.12.5-py2.py3-none-any.whl (147 kB)
     |################################| 147 kB 26.6 MB/s 
| Collecting chardet<5,>=3.0.2
|   Downloading chardet-4.0.0-py2.py3-none-any.whl (178 kB)
     |################################| 178 kB 18.3 MB/s 
| Collecting idna<3,>=2.5
|   Downloading idna-2.10-py2.py3-none-any.whl (58 kB)
     |################################| 58 kB 9.2 MB/s 
| Collecting urllib3<1.27,>=1.21.1
|   Downloading urllib3-1.26.3-py2.py3-none-any.whl (137 kB)
     |################################| 137 kB 15.4 MB/s 
| Collecting wrapt<2,>=1.10
|   Downloading wrapt-1.12.1.tar.gz (27 kB)
| Using legacy 'setup.py install' for python-levenshtein, since package 'wheel' is not installed.
| Using legacy 'setup.py install' for wrapt, since package 'wheel' is not installed.
| Installing collected packages: zipp, typing-extensions, pyparsing, importlib-metadata, wrapt, urllib3, toml, six, py, pluggy, packaging, iniconfig, idna, chardet, certifi, attrs, soupsieve, requests, pytz, python-levenshtein, python-dateutil, pytest, pyjwt, pillow, numpy, kiwisolver, fuzzywuzzy, deprecated, cycler, apipkg, typed-ast, tqdm, scipy, pytest-forked, pygithub, pyarrow, pandas, mypy-extensions, matplotlib, lxml, execnet, coverage, beautifulsoup4, pytest-xdist, pytest-cov, pybaseball, mypy
|     Running setup.py install for wrapt ... done
|     Running setup.py install for python-levenshtein ... done
|   Running setup.py develop for pybaseball
| Successfully installed apipkg-1.5 attrs-20.3.0 beautifulsoup4-4.9.3 certifi-2020.12.5 chardet-4.0.0 coverage-5.4 cycler-0.10.0 deprecated-1.2.11 execnet-1.8.0 fuzzywuzzy-0.18.0 idna-2.10 importlib-metadata-3.4.0 iniconfig-1.1.1 kiwisolver-1.3.1 lxml-4.6.2 matplotlib-3.3.4 mypy-0.800 mypy-extensions-0.4.3 numpy-1.19.5 packaging-20.9 pandas-1.1.5 pillow-8.1.0 pluggy-0.13.1 py-1.10.0 pyarrow-3.0.0 pybaseball pygithub-1.54.1 pyjwt-1.7.1 pyparsing-2.4.7 pytest-6.2.2 pytest-cov-2.11.1 pytest-forked-1.3.0 pytest-xdist-2.2.1 python-dateutil-2.8.1 python-levenshtein-0.12.2 pytz-2021.1 requests-2.25.1 scipy-1.5.4 six-1.15.0 soupsieve-2.2 toml-0.10.2 tqdm-4.56.2 typed-ast-1.4.2 typing-extensions-3.7.4.3 urllib3-1.26.3 wrapt-1.12.1 zipp-3.4.0
[PyTest On PR/Install dependencies and test]   ✅  Success - Install dependencies
[PyTest On PR/Install dependencies and test] ⭐  Run Run tests
| pytest tests -n 5 --doctest-modules --cov=pybaseball --cov-report term-missing
| ============================= test session starts ==============================
| platform linux -- Python 3.6.10, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
| rootdir: /github/workspace, configfile: pytest.ini, testpaths: tests
| plugins: forked-1.3.0, xdist-2.2.1, cov-2.11.1
gw0 [546] / gw1 [546] / gw2 [546] / gw3 [546] / gw4 [546]
| ........................................................................ [ 13%]
| ........................................................................ [ 26%]
| ........................................................................ [ 39%]
| ........................................................................ [ 52%]
| ........................................................................ [ 65%]
| ........................................................................ [ 79%]
| ........................................................................ [ 92%]
| ..........................................                               [100%]
| =============================== warnings summary ===============================
| tests/integration/pybaseball/test_statcast.py::test_handle_request_post_season
|   /github/workspace/pybaseball/statcast.py:45: UserWarning: 
|   That's a nice request you got there. It'd be a shame if something were to happen to it.
|   We strongly recommend that you enable caching before running this. It's as simple as `pybaseball.cache.enable()`.
|   Since the Statcast requests can take a *really* long time to run, if something were to happen, like: a disconnect;
|   gremlins; computer repair by associates of Rudy Giuliani; electromagnetic interference from metal trash cans; etc.;
|   you could lose a lot of progress. Enabling caching will allow you to immediately recover all the successful
|   subqueries if that happens.
|     warnings.warn(_OVERSIZE_WARNING)
| 
| -- Docs: https://docs.pytest.org/en/stable/warnings.html
| 
| ---------- coverage: platform linux, python 3.6.10-final-0 -----------
| Name                                                          Stmts   Miss  Cover   Missing
| -------------------------------------------------------------------------------------------
| pybaseball/__init__.py                                           72      0   100%
| pybaseball/amateur_draft.py                                      28      0   100%
| pybaseball/analysis/__init__.py                                   0      0   100%
| pybaseball/analysis/projections/__init__.py                       3      0   100%
| pybaseball/analysis/projections/marcels/__init__.py               3      0   100%
| pybaseball/analysis/projections/marcels/age_adjustment.py        11      2    82%   14, 19
| pybaseball/analysis/projections/marcels/marcels_base.py          85      3    96%   54, 57, 60
| pybaseball/analysis/projections/marcels/marcels_batting.py       20      0   100%
| pybaseball/analysis/projections/marcels/marcels_pitching.py      23      0   100%
| pybaseball/analysis/trajectories/__init__.py                      2      0   100%
| pybaseball/analysis/trajectories/batted_balls/__init__.py         0      0   100%
| pybaseball/analysis/trajectories/batted_balls/calculator.py      68      0   100%
| pybaseball/analysis/trajectories/batted_balls/parameters.py      32      0   100%
| pybaseball/analysis/trajectories/unit_conversions.py              5      0   100%
| pybaseball/analysis/trajectories/utils.py                        19      0   100%
| pybaseball/batting_leaders.py                                     5      0   100%
| pybaseball/cache/__init__.py                                      4      0   100%
| pybaseball/cache/cache.py                                        68      0   100%
| pybaseball/cache/cache_config.py                                 38      0   100%
| pybaseball/cache/cache_record.py                                 48      0   100%
| pybaseball/cache/dataframe_utils.py                              14      0   100%
| pybaseball/cache/file_utils.py                                   17      0   100%
| pybaseball/cache/func_utils.py                                    5      0   100%
| pybaseball/datahelpers/__init__.py                                0      0   100%
| pybaseball/datahelpers/column_mapper.py                          23      0   100%
| pybaseball/datahelpers/postprocessing.py                         92      6    93%   79-80, 133, 192-193, 205
| pybaseball/datahelpers/singleton.py                               8      0   100%
| pybaseball/datahelpers/statcast_utils.py                          9      9     0%   1-20
| pybaseball/datahelpers/transform.py                              11      1    91%   10
| pybaseball/datasources/__init__.py                                0      0   100%
| pybaseball/datasources/fangraphs.py                             112      4    96%   31-32, 124, 135
| pybaseball/datasources/html_table_processor.py                   42      2    95%   26, 74
| pybaseball/datasources/statcast.py                               17      0   100%
| pybaseball/enums/__init__.py                                      0      0   100%
| pybaseball/enums/enum_base.py                                    30      3    90%   16, 32, 43
| pybaseball/enums/fangraphs/__init__.py                           22      1    95%   23
| pybaseball/enums/fangraphs/batting_data_enum.py                 357      0   100%
| pybaseball/enums/fangraphs/fangraphs_stats_base.py               27      1    96%   37
| pybaseball/enums/fangraphs/fangraphs_stats_category.py            7      0   100%
| pybaseball/enums/fangraphs/fielding_data_enum.py                 88      0   100%
| pybaseball/enums/fangraphs/league.py                              8      0   100%
| pybaseball/enums/fangraphs/month.py                              13      0   100%
| pybaseball/enums/fangraphs/pitching_data_enum.py                381      0   100%
| pybaseball/enums/fangraphs/positions.py                          17      0   100%
| pybaseball/lahman.py                                             83     26    69%   22, 30-33, 54, 57, 60, 63, 66, 69, 75, 78, 84, 87, 90, 93, 96, 99, 102, 106, 115, 118, 121, 124, 133
| pybaseball/league_batting_stats.py                               56      1    98%   51
| pybaseball/league_pitching_stats.py                              61      1    98%   51
| pybaseball/pitching_leaders.py                                    5      0   100%
| pybaseball/playerid_lookup.py                                    80     18    78%   28-29, 48, 125-130, 143-158, 191-192, 204-205
| pybaseball/plotting.py                                          116     53    54%   55, 91-135, 155, 177, 181, 209-245, 256-262
| pybaseball/retrosheet.py                                        133    104    22%   118-148, 154-175, 181-205, 211-214, 220-232, 238-250, 257-260, 267-270, 277-280, 287-290, 297-300
| pybaseball/split_stats.py                                       116    105     9%   14-20, 27-35, 43-70, 78-170
| pybaseball/standings.py                                          79      0   100%
| pybaseball/statcast.py                                           44      0   100%
| pybaseball/statcast_batter.py                                    20      0   100%
| pybaseball/statcast_pitcher.py                                   11      0   100%
| pybaseball/team_batting.py                                       36     27    25%   24-59
| pybaseball/team_fielding.py                                      43     33    23%   26-84
| pybaseball/team_game_logs.py                                     31     22    29%   11-19, 23-36, 49-53
| pybaseball/team_pitching.py                                      37     28    24%   24-60
| pybaseball/team_results.py                                       89      3    97%   64-65, 113
| pybaseball/teamid_lookup.py                                      57      3    95%   17, 102, 108
| pybaseball/top_prospects.py                                      23     17    26%   9-23, 27-28
| pybaseball/utils.py                                             121     27    78%   102, 106, 131, 135, 162, 180, 218-220, 228-231, 244-270
| -------------------------------------------------------------------------------------------
| TOTAL                                                          3075    500    84%
| 
| ================== 546 passed, 1 warning in 116.06s (0:01:56) ==================
[PyTest On PR/Install dependencies and test]   ✅  Success - Run tests
[PyTest On PR/Install dependencies and test] ⭐  Run Run MyPy
| mypy --ignore-missing-imports --follow-imports silent *.py pybaseball tests
| pybaseball/datahelpers/column_mapper.py:7: error: Function is missing a return type annotation
| pybaseball/datahelpers/column_mapper.py:7: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_column_mapper.py:5: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_column_mapper.py:5: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_column_mapper.py:17: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_column_mapper.py:17: note: Use "-> None" if function does not return a value
| tests/pybaseball/analysis/projections/marcels/test_age.py:5: error: Function is missing a return type annotation
| tests/pybaseball/analysis/projections/marcels/test_age.py:5: note: Use "-> None" if function does not return a value
| pybaseball/enums/enum_base.py:18: error: Returning Any from function declared to return "_T"
| pybaseball/enums/enum_base.py:30: error: Returning Any from function declared to return "Optional[_T]"
| pybaseball/datahelpers/postprocessing.py:191: error: Function is missing a type annotation
| pybaseball/datahelpers/postprocessing.py:196: error: Function is missing a type annotation
| pybaseball/datahelpers/postprocessing.py:203: error: Function is missing a type annotation
| pybaseball/analysis/trajectories/batted_balls/calculator.py:71: error: Returning Any from function declared to return "float"
| pybaseball/analysis/trajectories/batted_balls/calculator.py:100: error: Returning Any from function declared to return "float"
| pybaseball/split_stats.py:31: error: Incompatible types in assignment (expression has type "str", variable has type "Optional[int]")
| pybaseball/split_stats.py:141: error: "List[List[Any]]" has no attribute "rename"
| pybaseball/split_stats.py:141: error: "List[List[Any]]" has no attribute "iloc"
| pybaseball/split_stats.py:142: error: "List[List[Any]]" has no attribute "reindex"; maybe "index"?
| pybaseball/split_stats.py:142: error: "Callable[[List[Any], int, int], int]" has no attribute "drop"
| pybaseball/split_stats.py:143: error: "List[List[Any]]" has no attribute "set_index"
| pybaseball/split_stats.py:144: error: "List[List[Any]]" has no attribute "drop"
| pybaseball/split_stats.py:145: error: "List[List[Any]]" has no attribute "apply"
| pybaseball/split_stats.py:146: error: "List[List[Any]]" has no attribute "dropna"
| pybaseball/split_stats.py:147: error: No overload variant of "__setitem__" of "list" matches argument types "str", "Any"
| pybaseball/split_stats.py:147: note: Possible overload variants:
| pybaseball/split_stats.py:147: note:     def __setitem__(self, int, List[Any]) -> None
| pybaseball/split_stats.py:147: note:     def __setitem__(self, slice, Iterable[List[Any]]) -> None
| pybaseball/split_stats.py:147: error: No overload variant of "__getitem__" of "list" matches argument type "str"
| pybaseball/split_stats.py:147: note:     def __getitem__(self, int) -> List[Any]
| pybaseball/split_stats.py:147: note:     def __getitem__(self, slice) -> List[List[Any]]
| pybaseball/split_stats.py:148: error: "List[List[Any]]" has no attribute "loc"
| pybaseball/split_stats.py:151: error: "List[List[Any]]" has no attribute "rename"
| pybaseball/split_stats.py:151: error: "List[List[Any]]" has no attribute "iloc"
| pybaseball/split_stats.py:152: error: "List[List[Any]]" has no attribute "reindex"; maybe "index"?
| pybaseball/split_stats.py:152: error: "Callable[[List[Any], int, int], int]" has no attribute "drop"
| pybaseball/split_stats.py:153: error: "List[List[Any]]" has no attribute "set_index"
| pybaseball/split_stats.py:154: error: "List[List[Any]]" has no attribute "drop"
| pybaseball/split_stats.py:155: error: "List[List[Any]]" has no attribute "apply"
| pybaseball/split_stats.py:157: error: "List[List[Any]]" has no attribute "dropna"
| pybaseball/split_stats.py:158: error: "List[List[Any]]" has no attribute "loc"
| tests/pybaseball/datahelpers/test_postprocessing.py:11: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:22: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:22: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:25: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:25: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:29: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:29: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:33: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:33: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:36: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:36: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:39: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:39: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:42: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:42: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:45: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:45: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:48: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:48: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:51: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:51: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:54: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:54: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:57: error: Function is missing a return type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:57: note: Use "-> None" if function does not return a value
| tests/pybaseball/datahelpers/test_postprocessing.py:78: error: Function is missing a type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:91: error: Function is missing a type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:98: error: Function is missing a type annotation
| tests/pybaseball/datahelpers/test_postprocessing.py:114: error: Function is missing a type annotation
| tests/pybaseball/analysis/trajectories/batted_balls/test_parameters.py:6: error: Function is missing a return type annotation
| tests/pybaseball/analysis/trajectories/batted_balls/test_parameters.py:6: note: Use "-> None" if function does not return a value
| tests/pybaseball/analysis/trajectories/batted_balls/test_calculator.py:6: error: Function is missing a type annotation
| tests/pybaseball/analysis/trajectories/batted_balls/test_calculator.py:10: error: Function is missing a return type annotation
| tests/pybaseball/analysis/trajectories/batted_balls/test_calculator.py:10: note: Use "-> None" if function does not return a value
| tests/pybaseball/analysis/trajectories/batted_balls/test_calculator.py:14: error: Function is missing a return type annotation
| tests/pybaseball/analysis/trajectories/batted_balls/test_calculator.py:14: note: Use "-> None" if function does not return a value
| tests/pybaseball/analysis/trajectories/batted_balls/test_calculator.py:27: error: Function is missing a return type annotation
| tests/pybaseball/analysis/trajectories/batted_balls/test_calculator.py:27: note: Use "-> None" if function does not return a value
| tests/pybaseball/analysis/trajectories/batted_balls/test_batted_balls_utils.py:9: error: Function is missing a type annotation
| tests/pybaseball/analysis/trajectories/batted_balls/test_batted_balls_utils.py:22: error: Function is missing a type annotation
| pybaseball/utils.py:177: error: Function is missing a type annotation
| pybaseball/utils.py:214: error: Function is missing a type annotation
| pybaseball/utils.py:223: error: Function is missing a type annotation
| pybaseball/top_prospects.py:8: error: Function is missing a type annotation
| pybaseball/top_prospects.py:26: error: Function is missing a type annotation
| pybaseball/team_game_logs.py:10: error: Function is missing a type annotation
| pybaseball/team_game_logs.py:22: error: Function is missing a type annotation
| pybaseball/team_game_logs.py:40: error: Function is missing a type annotation
| pybaseball/playerid_lookup.py:19: error: Function is missing a return type annotation
| pybaseball/playerid_lookup.py:53: error: Function is missing a type annotation
| pybaseball/playerid_lookup.py:110: error: Argument "first" to "get_closest_names" has incompatible type "Optional[str]"; expected "str"
| pybaseball/lahman.py:28: error: Function is missing a return type annotation
| pybaseball/lahman.py:28: note: Use "-> None" if function does not return a value
| pybaseball/team_results.py:14: error: Function is missing a type annotation
| pybaseball/team_results.py:22: error: Function is missing a type annotation
| pybaseball/team_results.py:74: error: Function is missing a type annotation
| pybaseball/team_results.py:86: error: Function is missing a type annotation
| pybaseball/team_results.py:101: error: Function is missing a type annotation
| pybaseball/team_pitching.py:15: error: Function is missing a type annotation
| pybaseball/team_fielding.py:16: error: Function is missing a type annotation
| pybaseball/team_batting.py:15: error: Function is missing a type annotation
| pybaseball/retrosheet.py:109: error: Function is missing a type annotation
| pybaseball/retrosheet.py:150: error: Function is missing a type annotation
| pybaseball/retrosheet.py:177: error: Function is missing a type annotation
| pybaseball/retrosheet.py:207: error: Function is missing a return type annotation
| pybaseball/retrosheet.py:216: error: Function is missing a type annotation
| pybaseball/retrosheet.py:234: error: Function is missing a type annotation
| pybaseball/retrosheet.py:253: error: Function is missing a return type annotation
| pybaseball/retrosheet.py:263: error: Function is missing a return type annotation
| pybaseball/retrosheet.py:273: error: Function is missing a return type annotation
| pybaseball/retrosheet.py:283: error: Function is missing a return type annotation
| pybaseball/retrosheet.py:293: error: Function is missing a return type annotation
| pybaseball/league_pitching_stats.py:12: error: Function is missing a type annotation
| pybaseball/league_pitching_stats.py:22: error: Function is missing a type annotation
| pybaseball/league_pitching_stats.py:40: error: Function is missing a type annotation
| pybaseball/league_pitching_stats.py:71: error: Function is missing a type annotation
| pybaseball/league_pitching_stats.py:84: error: Function is missing a type annotation
| tests/pybaseball/test_team_pitching.py:12: error: Returning Any from function declared to return "str"
| tests/pybaseball/test_team_pitching.py:20: error: Function is missing a return type annotation
| tests/pybaseball/test_team_fielding.py:12: error: Returning Any from function declared to return "str"
| tests/pybaseball/test_team_fielding.py:20: error: Function is missing a return type annotation
| tests/pybaseball/test_team_batting.py:12: error: Returning Any from function declared to return "str"
| tests/pybaseball/test_team_batting.py:20: error: Function is missing a return type annotation
| tests/pybaseball/test_plotting.py:22: error: Incompatible types in assignment (expression has type "None", target has type Module)
| tests/pybaseball/test_plotting.py:28: error: Module has no attribute "Chart"
| tests/pybaseball/test_plotting.py:29: error: Module has no attribute "Color"
| tests/pybaseball/test_plotting.py:30: error: Module has no attribute "LayerChart"
| tests/pybaseball/test_plotting.py:31: error: Module has no attribute "Scale"
| tests/pybaseball/test_plotting.py:32: error: Module has no attribute "X"
| tests/pybaseball/test_plotting.py:33: error: Module has no attribute "Y"
| tests/pybaseball/test_plotting.py:50: error: Function is missing a type annotation for one or more arguments
| tests/pybaseball/test_pitching_leaders.py:11: error: Returning Any from function declared to return "str"
| tests/pybaseball/test_pitching_leaders.py:19: error: Function is missing a return type annotation
| tests/pybaseball/test_batting_leaders.py:20: error: Function is missing a return type annotation
| tests/pybaseball/profiling.py:19: error: Function is missing a return type annotation
| tests/pybaseball/profiling.py:38: error: Function is missing a return type annotation
| tests/pybaseball/profiling.py:39: error: Function is missing a return type annotation
| tests/pybaseball/profiling.py:39: note: Use "-> None" if function does not return a value
| tests/pybaseball/profiling.py:47: error: Function is missing a return type annotation
| tests/pybaseball/profiling.py:47: note: Use "-> None" if function does not return a value
| tests/pybaseball/profiling.py:51: error: Function is missing a return type annotation
| tests/pybaseball/profiling.py:51: note: Use "-> None" if function does not return a value
| tests/pybaseball/profiling.py:55: error: Function is missing a return type annotation
| tests/pybaseball/profiling.py:55: note: Use "-> None" if function does not return a value
| tests/pybaseball/profiling.py:59: error: Function is missing a return type annotation
| tests/pybaseball/profiling.py:59: note: Use "-> None" if function does not return a value
| tests/pybaseball/profiling.py:63: error: Function is missing a return type annotation
| tests/pybaseball/profiling.py:63: note: Use "-> None" if function does not return a value
| tests/pybaseball/profiling.py:76: error: Function is missing a return type annotation
| tests/integration/pybaseball/datasources/test_fg_pitching_data.py:65: error: Function is missing a type annotation for one or more arguments
| tests/integration/pybaseball/datasources/test_fg_pitching_data.py:90: error: Function is missing a type annotation for one or more arguments
| tests/integration/pybaseball/datasources/test_fg_pitching_data.py:123: error: Function is missing a type annotation for one or more arguments
| pybaseball/datahelpers/transform.py:8: error: Function is missing a type annotation
| Found 122 errors in 31 files (checked 125 source files)
| Makefile:27: recipe for target 'mypy' failed
| make: *** [mypy] Error 1
[PyTest On PR/Install dependencies and test]   ❌  Failure - Run MyPy
Error: exit with `FAILURE`: 2
```

</details>

--- a/pkg/runner/run_context.go
+++ b/pkg/runner/run_context.go
@@ -225,7 +225,13 @@ func (rc *RunContext) newStepExecutor(step *model.Step) common.Executor {
 			common.Logger(ctx).Infof("  \u2705  Success - %s", sc.Step)
 		} else {
 			common.Logger(ctx).Errorf("  \u274C  Failure - %s", sc.Step)
-			rc.StepResults[rc.CurrentStep].Success = false
+
+			if sc.Step.ContinueOnError {
+				log.Infof("Failed but conitnue next step")
+				err = nil
+			} else {
+				rc.StepResults[rc.CurrentStep].Success = false
+			}
 		}
 		return err
 	}

--- a/pkg/runner/run_context.go
+++ b/pkg/runner/run_context.go
@@ -227,8 +227,9 @@ func (rc *RunContext) newStepExecutor(step *model.Step) common.Executor {
 			common.Logger(ctx).Errorf("  \u274C  Failure - %s", sc.Step)
 
 			if sc.Step.ContinueOnError {
-				log.Infof("Failed but conitnue next step")
+				common.Logger(ctx).Infof("Failed but continue next step")
 				err = nil
+				rc.StepResults[rc.CurrentStep].Success = true
 			} else {
 				rc.StepResults[rc.CurrentStep].Success = false
 			}

